### PR TITLE
Remove notifications for instances no longer referenced in deployment.xml

### PR DESCRIPTION
**Pull request by freva**

### CODE REVIEW

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/ApplicationController.java [hunk 1]
```diff
@@ -445,6 +445,11 @@ public LockedApplication storeWithUpdatedConfig(LockedApplication application, A
         // Validate new deployment spec thoroughly before storing it.
         controller.jobController().deploymentStatus(application.get());
 
+        // Clear notifications for instances that are no longer declared
+        for (var name : existingInstances)
+            if ( ! declaredInstances.contains(name))
+                controller.notificationsDb().removeNotifications(NotificationSource.from(application.get().id().instance(name)));
```
Comment (jonmv) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/ApplicationController.java:451]: We should probably also do the other cleanup, like removing DNS entries?

**Review by jonmv — APPROVED** [review]
Nice, more! :p

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/ApplicationController.java (hunk 1 above)
Comment (freva) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/ApplicationController.java:451]: Yes, I'm not sure what happens to those now and if its safe to do it here? FYI: @mpolden

Comment (mpolden) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/ApplicationController.java:451]: `deleteInstance(ApplicationId)` already removes DNS records.

Comment (freva) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/ApplicationController.java:451]: Yes... The problem here is that when an instance is removed from `deployment.xml`, `deleteInstance(ApplicationId)` is never called.

Comment (mpolden) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/ApplicationController.java:451]: 😞 we should make it one shared code path.

**Pull request merged**